freeipa-client/src/main/java/com/sequenceiq/freeipa/client/operation/GroupAddOperation.java

```diff
@@ -40,9 +47,25 @@ public String getOperationName() {
         return List.of(group);
     }
 
+    @Override
+    protected Map<String, Object> getParams() {
+        Map<String, Object> map = new HashMap<>();
+        switch (groupType) {
+            case NONPOSIX:
+                map.put("nonposix", true);
+                break;
+            case POSIX:
+            default:
+                // no extra parameters
+                break;
+        }
+        return ImmutableMap.copyOf(map);
```

Comment:
why is the immutability important here? As currently there is one field only in one case it could be simplified as:
```
        switch (groupType) {
            case NONPOSIX:
               return Map.of("nonposix", true);
            case POSIX:
            default:
		return Map.of();
```
But I'm ok with the `HashMap` too, just don't get the ImmutableMap part

Reply:
I had the hashmap so that we can have a single return statement from this method. The method is simple enough that returning from the switch is still understandable.
I pulled the maps out to constants since the params are not dependent upon the group name and won't change across operations.